Doc/tutorial/datastructures.rst

```diff
@@ -497,7 +497,9 @@ You can't use lists as keys, since lists can be modified in place using index
 assignments, slice assignments, or methods like :meth:`append` and
 :meth:`extend`.
 
-It is best to think of a dictionary as an unordered set of *key: value* pairs,
+Dictionaries are guaranteed to retain insertion order, not ordered by key.
```

Comment:
Just remove `unordered`, and don't mention about ordering here.
While dict preserves insertion order, it's not so important in "What is dict?".
For example, `dict.__eq__` ignores order.